### PR TITLE
[FEATURE] Get profilers from DataContext

### PR DESCRIPTION
**Pull request by cdkini**
Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html).

Changes proposed in this pull request:
- Get method and related tests

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/en/latest/contributing/style_guide.html?highlight=style%20guide)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html?highlight=checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/en/latest/contributing/testing.html#contributing-testing-writing-unit-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: great_expectations/checkpoint/toolkit.py [hunk 1]
```diff
@@ -166,10 +166,6 @@ def get_checkpoint(
         raise ge_exceptions.CheckpointNotFoundError(
             message=f'Non-existent Checkpoint configuration named "{key.configuration_key}".\n\nDetails: {exc_ik}'
         )
-    except ValidationError as exc_ve:
-        raise ge_exceptions.InvalidCheckpointConfigError(
-            message="Invalid Checkpoint configuration", validation_error=exc_ve
-        )
```
Comment (cdkini) [inline great_expectations/checkpoint/toolkit.py]: This feature mirrors the functionality from Checkpoint - this ValidationError has no place in this method as we are only retrieving values from the store (not running schema checks). This check is absent from the Profiler implementation.

File: great_expectations/rule_based_profiler/profiler.py [hunk 2]
```diff
@@ -7,12 +7,7 @@
 from great_expectations.core.expectation_configuration import ExpectationConfiguration
 from great_expectations.core.expectation_suite import ExpectationSuite
 from great_expectations.data_context.util import instantiate_class_from_config
-from great_expectations.rule_based_profiler.config.base import (
-    RuleConfig,
-    domainBuilderConfigSchema,
-    expectationConfigurationBuilderConfigSchema,
-    parameterBuilderConfigSchema,
```
Comment (cdkini) [inline great_expectations/rule_based_profiler/profiler.py]: Unused imports from a prior commit

File: tests/rule_based_profiler/conftest.py [hunk 3]
```diff
@@ -2404,3 +2417,73 @@ def rule_with_variables_with_parameters(
...
     }
     return rule
+
+
+# Fixtures from config_version 1.0 ============================================================================
+
+
+@pytest.fixture(scope="function")
+def profiler_name() -> str:
+    return "my_first_profiler"
```
Comment (cdkini) [inline tests/rule_based_profiler/conftest.py]: These fixtures have wider applicability than their origin file so moving to conftest to reuse.

File: great_expectations/rule_based_profiler/config/base.py [hunk 4]
```diff
@@ -51,21 +51,36 @@ def make_config(self, data: dict, **kwargs) -> Type[DictDot]:
 
         return self.__config_class__(**data)
 
-    @post_dump
-    def remove_nulls(self, data: dict, **kwargs) -> dict:
+    @post_dump(pass_original=True)
+    def remove_nulls_and_keep_unknowns(
```
Comment (cdkini) [inline great_expectations/rule_based_profiler/config/base.py:55]: This is a very much needed piece of Marshmallow magic.

Since we use `setattr` to dynamically set values, the schema isn't aware of all possible types the object can have. When writing to a dictionary using `dump`, the config object validates against the schema and then serializes, implicitly removing unknown values as part of the process.

We want to persist these values - as such, we use the `pass_original=True` argument in the decorator to gain access to the original config object. From here, we can ensure that we're persisting this unknown values in the resulting dictionary object.

One caveat is that we need to manually filter our private attributes (like `_commented_map`) to get this working. Perhaps this is a little hacky but it follows Marshmallow conventions and is the cleanest way to enable proper serialization when utilizing our `setattr` strategy.

TLDR: validated on load and partially validated on dump (ignoring dynamic `setattr` attributes)

File: great_expectations/data_context/data_context.py [hunk 5]
```diff
@@ -3249,6 +3251,18 @@ def run_checkpoint(
...
+        name: Optional[str] = None,
+        ge_cloud_id: Optional[str] = None,
+    ) -> RuleBasedProfiler:
+        return profiler_toolkit.get_profiler(
+            data_context=self,
+            profiler_store=self.profiler_store,
+            name=name,
+            ge_cloud_id=ge_cloud_id,
+        )
+
```
Comment (cdkini) [inline great_expectations/data_context/data_context.py:3265]: @donaldheppner I know we want to move these `toolkit` utilities to an actual class.

Two questions:
* Where do you think that logically falls in our class hierarchy?
* Could that be part of a following PR to refactor the structure of RBP?

**Review by alexsherstinsky — APPROVED** [review]
LGTM
